src/ProxyManager/Generator/MethodGenerator.php

```diff
@@ -19,14 +20,27 @@ class MethodGenerator extends LaminasMethodGenerator
     public static function fromReflectionWithoutBodyAndDocBlock(MethodReflection $reflectionMethod): self
     {
         /** @var static $method */
-        $method = parent::copyMethodSignature($reflectionMethod);
+        $method = static::copyMethodSignature($reflectionMethod);
 
         $method->setInterface(false);
         $method->setBody('');
 
         return $method;
     }
 
+    public static function copyMethodSignature(MethodReflection $reflectionMethod): parent
+    {
+        $method = parent::copyMethodSignature($reflectionMethod);
+
+        foreach ($reflectionMethod->getParameters() as $reflectionParameter) {
+            $method->setParameter(
+                ParameterGenerator::fromReflection($reflectionParameter)
+            );
+        }
+
+        return $method;
```

Comment:
Shouldn't this behavior be fixed in `laminas/laminas-code`, if we're required to copy parameter declarations twice to fix it?